plugins/repository-azure/src/main/java/org/opensearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -246,7 +259,8 @@ private AzureStorageSettings(
         ProxySettings proxySettings
     ) {
         this.account = account;
-        this.connectString = buildConnectString(account, key, sasToken, endpointSuffix);
+        this.connectString = buildConnectString(account, key, sasToken, endpointSuffix, tokenCredentialType);
+        this.tokenCredentialType = validateTokenCredentialType(tokenCredentialType);
```

Comment:
Not needed, the value will be validated by the setting:
```suggestion
        this.tokenCredentialType = tokenCredentialType;
```

Reply:
refactored